pkg/workload/kv/kv.go

```diff
@@ -942,40 +965,90 @@ func (g *sequentialGenerator) state() string {
 	return fmt.Sprintf("S%d", g.seq.read())
 }
 
-type zipfGenerator struct {
-	seq    *sequence
+type writeSequenceSource struct {
+	*sequence
+}
+
+func (wss writeSequenceSource) Int63() int64 {
+	return int64(wss.Uint64() & 0x7fffffffffffffff) // Masks high bit
+}
+
+func (wss writeSequenceSource) Uint64() uint64 {
+	hasher := crc64.New(crc64.MakeTable(crc64.ECMA))
+	buf := []byte{0, 0, 0, 0, 0, 0, 0, 0}
+	// only increment the sequence during write calls
+	binary.BigEndian.PutUint64(buf[:8], uint64(wss.write()))
+	_, _ = hasher.Write(buf[:8])
+	return hasher.Sum64()
+}
+
+func (wss writeSequenceSource) Seed(seed int64) {
+	wss.val.Store(seed)
+}
+
+type readSequenceSource struct {
+	*sequence
 	random *rand.Rand
-	zipf   *zipf
 }
 
-// Creates a new zipfian generator.
-func newZipfianGenerator(seq *sequence, rng *rand.Rand) *zipfGenerator {
-	return &zipfGenerator{
-		seq:    seq,
-		random: rng,
-		zipf:   newZipf(1.1, 1, uint64(math.MaxInt64)),
+func (rss readSequenceSource) Int63() int64 {
+	return int64(rss.Uint64() & 0x7fffffffffffffff) // Masks high bit
+}
+
+func (rss readSequenceSource) Uint64() uint64 {
+	hasher := crc64.New(crc64.MakeTable(crc64.ECMA))
```

Comment:
Same as above. Consider reusing the `hasher`.

Reply:
Is `hasher` isn't reentrant and has state: https://cs.opensource.google/go/go/+/refs/tags/go1.24.1:src/hash/crc64/crc64.go;l=87-91

The above use of sha1 in this program is, I believe, incorrect for this very same reason, but I haven't confirmed.